lib/qmp.ml

```diff
@@ -150,6 +170,12 @@ let message_of_string x =
           let service = int_of_string (string (List.assoc "service" list)) in
           let host = string (List.assoc "host" list) in
           {enabled; auth; family; service; host})
+      | `Assoc list when List.mem_assoc "product-num" list
+                      && List.mem_assoc "build-num" list ->
+        Xen_platform_pv_driver_info (
+          let product_num = int (List.assoc "product-num" list) in
```

Comment:
What happens where the entries are not numbers as expected? Could you add that as a test case?